commands/poll.js

```diff
@@ -0,0 +1,333 @@
+          .setLabel("Close Poll")
+          .setStyle(ButtonStyle.Danger)
+      );
+
+      const pollMessage = await interaction.reply({
+        embeds: [embed],
+        components: [row],
+        fetchReply: true,
+      });
+
+      for (let i = 0; i < options.length; ++i) {
+        await pollMessage.react(`${OPTION_EMOJIS[i]}`);
+      }
+
+      const reactionFilter = (reaction, user) => {
+        return OPTION_EMOJIS.includes(reaction.emoji.name) && !user.bot;
+      };
+
+      const reactionCollector = pollMessage.createReactionCollector({
+        reactionFilter,
+        time: pollCloseInSeconds * 1000,
+        dispose: true,
+      });
+
+      reactionCollector.on("collect", async (reaction, user) => {
+        console.info(`Got reaction ${reaction.emoji.name}  from ${user.tag}`);
+      });
+
+      reactionCollector.on("remove", async (reaction, user) => {
+        console.info(`Removed reaction ${reaction.emoji.name}  by ${user.tag}`);
+      });
+
+      reactionCollector.on("end", (collected) => {
+        console.debug(`Collected ${collected.size} reaction interactions!`);
+      });
+
+      const buttonCollector = pollMessage.createMessageComponentCollector({
+        componentType: ComponentType.Button,
+        time: pollCloseInSeconds * 1000,
+      });
+
+      buttonCollector.on("collect", async (action) => {
+        if (action.user.id === interaction.member.user.id) {
+          const originalEmbed = pollMessage.embeds[0];
+          const computingResultsEmbed = {
+            title: "Computing poll results...",
+            footer: {
+              text: "This bot is powered by Xander's money",
+            },
+          };
+
+          await pollMessage.edit({
+            embeds: [computingResultsEmbed],
+            components: [],
+          });
+
+          const finalResults = await getTopVotedOption(
+            pollMessage,
+            interaction
+          );
+
+          const finalEmbed = EmbedBuilder.from(originalEmbed)
+            .setDescription(
+              finalResults.length > 1
+                ? `Closed by ${interaction.member.user.tag}, see top options below!`
+                : `Closed by ${interaction.member.user.tag}, see top option below!`
+            )
+            .spliceFields(0, options.length)
+            .addFields(
+              finalResults.map((result) => {
+                const option = options.find(
+                  (item) => item.emoji === result.emoji
+                );
+
+                return {
+                  name:
+                    result.count > 1
+                      ? `${option.emoji}  - ${option.option} (${result.count} votes)`
+                      : `${option.emoji}  - ${option.option} (1 vote)`,
+                  value:
+                    result.count > 0
+                      ? `Voted by: ${result.reactors}`
+                      : `Voted by no one...`,
+                };
+              })
+            );
```

Comment:
Suggestion: Although the description would differ depending on whether the poll was closed by the author or was closed automatically after a certain period of time, it might be cleaner to create a constant function that would return this. 

You'll just pass the `originalEmbed`, `description`, and `finalResults`.

Also, instead of using `spliceFields` and then `addFields`, you might wanna try using `setFields` instead. Since according to this [doc](https://discordjs.guide/additional-info/changes-in-v13.html#messageembed-1), `embed.setFields(newFields)` is equivalent to `embed.spliceFields(0, embed.fields.length, newFields)`.